### ISSUE TITLE
Question: ArchiveReader::read() blocks?

### ISSUE THREAD

**Issue by john-terrell** [rust-compress/rc-zip#10]
Hi, this is a question and not really an issue.   I'm looking at using rc-zip with in an async-await environment.   However, ArchiveReader::read() looks to be the only way to feed the reader file data and it will block during the read.   Is there an alternate way to feed it data that I may have read using async read directly?   Is an async version of read() necessary?  

Thanks for clarifying.

**Comment by fasterthanlime** [comment]
So, first off, rc-zip is in a semi-incomplete state and I don't recommend
using it in anything serious at this stage.

That said - it is designed as a state machine, specifically so it can adapt
to different I/O model. The 'read' method you mention implement a
synchronous model on top of it but you can implement an async model as
well. It hasn't been done yet because async/await wasn't stable back then!

tl;dr this is in the spirit of the project but the project is only half
baked

**Comment by john-terrell** [comment]
Thanks for the response.   I have a project that needs the functionality provided by rc-zip so I may fork the repo and work on adding async to it.   I'll send a PR when it's ready if you'd like.

**Comment by fasterthanlime** [comment]
Sure, I'd be happy to review changes! Maybe @Geal will be able to help too (he's in the rust-compress org), but I don't know what his availability is like these days. Mine is spotty, so expect delays :)
